sql/catalyst/src/main/scala/org/apache/spark/sql/types/DecimalType.scala

```diff
@@ -109,6 +109,7 @@ object DecimalType extends AbstractDataType {
   val MAX_SCALE = 38
   val SYSTEM_DEFAULT: DecimalType = DecimalType(MAX_PRECISION, 18)
   val USER_DEFAULT: DecimalType = DecimalType(10, 0)
+  val BIGINT_DEFAULT: DecimalType = DecimalType(MAX_PRECISION, 0)
```

Comment:
please add a `private[sql] val BigIntegerDecimal = DecimalType(38, 0)` to the next section, instead of doing this.